src/legacy/core_plugins/kibana/server/ui_setting_defaults.js

```diff
@@ -18,53 +18,32 @@
  */
 
 import moment from 'moment-timezone';
-import numeralLanguages from '@elastic/numeral/languages';
 import { i18n } from '@kbn/i18n';
 import { schema } from '@kbn/config-schema';
 
 import { isRelativeUrl } from '../../../../core/server';
-import { DEFAULT_QUERY_LANGUAGE } from '../../../../plugins/data/common';
 
 export function getUiSettingDefaults() {
   const weekdays = moment.weekdays().slice();
   const [defaultWeekday] = weekdays;
-  // We add the `en` key manually here, since that's not a real numeral locale, but the
-  // default fallback in case the locale is not found.
-  const numeralLanguageIds = [
-    'en',
-    ...numeralLanguages.map(function(numeralLanguage) {
-      return numeralLanguage.id;
-    }),
-  ];
-
-  const luceneQueryLanguageLabel = i18n.translate(
-    'kbn.advancedSettings.searchQueryLanguageLucene',
-    {
-      defaultMessage: 'Lucene',
-    }
-  );
-
-  const queryLanguageSettingName = i18n.translate('kbn.advancedSettings.searchQueryLanguageTitle', {
-    defaultMessage: 'Query language',
-  });
-
-  const requestPreferenceOptionLabels = {
-    sessionId: i18n.translate('kbn.advancedSettings.courier.requestPreferenceSessionId', {
-      defaultMessage: 'Session ID',
-    }),
-    custom: i18n.translate('kbn.advancedSettings.courier.requestPreferenceCustom', {
-      defaultMessage: 'Custom',
-    }),
-    none: i18n.translate('kbn.advancedSettings.courier.requestPreferenceNone', {
-      defaultMessage: 'None',
-    }),
-  };
 
   // wrapped in provider so that a new instance is given to each app/test
   return {
     buildNum: {
       readonly: true,
     },
+    'state:storeInSessionStorage': {
+      name: i18n.translate('kbn.advancedSettings.storeUrlTitle', {
+        defaultMessage: 'Store URLs in session storage',
+      }),
+      value: false,
+      description: i18n.translate('kbn.advancedSettings.storeUrlText', {
+        defaultMessage:
+          'The URL can sometimes grow to be too large for some browsers to handle. ' +
+          'To counter-act this we are testing if storing parts of the URL in session storage could help. ' +
+          'Please let us know how it goes!',
+      }),
+    },
```

Comment:
It looks like this one was flagged to go in `kibana_utils` in the original issue, but I'm wondering if we should put it in `data` for the time being instead?

`kibana_utils` doesn't really _do_ anything with core, in fact, it only has a plugin interface with lifecycles for bundling purposes.

WDYT @Dosant? Should the `storeInSessionStorage` setting live in data? Should we break our rule about `kibana_utils` being essentially static to have it registered from there instead? Or should we put it somewhere else?

Reply:
Let's consider that all the `storeInSessionStorage` will likely could be removed in 7.9? (As I understand this workaround exists only for IE11?). 

- Maybe this is a good topic to discuss within, I'll add to agenda on one of next syncs.

So with that in mind, I'd go for the simplest approach and would try to not introduce new cross plugin deps or making kibana_utils not static. And if it is not kibana_utils, then leaving it in `core` sounds better to me then moving it into `data`

Reply:
> Maybe this is a good topic to discuss within, I'll add to agenda on one of next syncs.

++ Let's definitely discuss this further. Deprecating it and removing with the rest of the legacy platform seems like a viable option, but we should talk through whether anything else needs to be done to ensure there isn't unexpected breakage.

Reply:
But to clarify, I agree let's leave it as-is for now